vyper/builtin_functions/functions.py

```diff
@@ -2505,6 +2506,48 @@ def build_IR(self, expr, args, kwargs, context):
             )
 
 
+class _MinMaxValue(BuiltinFunction):
+    _inputs = [("typename", "TYPE_DEFINITION")]
+
+    def evaluate(self, node):
+        self._validate_arg_types(node)
+        input_type = get_type_from_annotation(node.args[0], DataLocation.UNSET)
+
+        if not isinstance(input_type, NumericAbstractType):
+            raise InvalidType(f"Expected numeric type but got {input_type} instead", node)
+
+        if isinstance(input_type, DecimalDefinition):
+            val = self._eval(SizeLimits.MIN_AST_DECIMAL, SizeLimits.MAX_AST_DECIMAL)
+            return vy_ast.Decimal.from_node(node, value=val)
+
+        if isinstance(input_type, IntegerAbstractType):
+            (lo, hi) = int_bounds(input_type._is_signed, input_type._bits)
+            val = self._eval(lo, hi)
```

Comment:
it's nice to use this `_eval` function. dispatching to `min/max` is a bit magical. for instance, if the signature of `int_bounds` ever breaks, i think it would be confusing that this function doesn't break with it.

i would say this would be clearer with two functions, `_eval_decimal` and `_eval_int`. each takes the input type and returns a value. so this block would look more like
```
        if isinstance(input_type, IntegerAbstractType):
            val = self._eval_int(input_type)
```
and similar for the decimal case.